Assets/Scripts/Entity/Player/Sanity/SanityMath.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using UnityEngine;
+using Utils;
+using static Utils.Enums;
+
+namespace Entity.Player.Sanity
+{
+    /// <summary>
+    /// Handles the math for stats influencing the player's sanity.
+    /// </summary>
+    [Serializable]
+    public class SanityMath
+    {
+        [Tooltip("The total severity at which sanity changes. Scales the change rate")]
+        [SerializeField]
+        private float severity;
+
+        [Tooltip("If one or more stats have a negative impact, weight other stats with this percentage.")]
+        [SerializeField]
+        private float lowImpactSeverity;
+
+        [Tooltip("Determines how fast player gains sanity. Scales current stat value * stat severity")]
+        [SerializeField]
+        private float gainFactor;
+
+        [Tooltip("Stats that have an impact on sanity.")]
+        [SerializeField]
+        private Stat[] stats;
+
+
+        // the total change rate
+        private float _changeRate;
+
+        /// <summary>
+        /// Gets the current sanity change which should be added to the next tick.
+        /// </summary>
+        public float GetCurrentChange()
+        {
+            return _changeRate * severity;
+        }
+
+        /// <summary>
+        /// Influences sanity by stat and its current value. It changes the stat's rate. It will get negative
+        /// as the current stat value gets below the stat boundary. If it's positive it is weighted with a
+        /// gain factor additionally.
+        /// </summary>
+        /// <param name="statType">Type of stat</param>
+        /// <param name="value">Current value of stat</param>
+        public void InfluenceSanityByStat(StatType statType, int value)
+        {
+            Stat stat = GetStatByType(statType);
+            if (value < stat.Boundary)
+                stat.Rate = (value - stat.Boundary) * stat.Severity;
+            else
+                // don't use - stat.Boundary because then if you have 100% of stat,
+                // you get a higher positive rate, the lower the boundary
+                stat.Rate = value * gainFactor * stat.Severity;
```

Comment:
Probably more of a implementation question:
Should saturation and hydration influence the sanity positively? Or should only actions (e.g. sleeping) and/or using specific items (e.g. cooked food or better food) influence it that way?

Reply:
I think it's okay for it to be influenced by sufficient saturation/hydration, too

Reply:
Yep I agree